core/modal_models/modes.c

```diff
@@ -558,4 +557,51 @@ void _lf_terminate_modal_reactors() {
     }
     _lf_unsused_suspended_events_head = NULL;
 }
+
+/**
+ * @brief This function accepts an array of ALL the mode_states and should add a reactor ptr into return_vec 
```

Comment:
```suggestion
 * @brief This function accepts an array of ALL the mode_states and adds a reactor ptr into return_vec 
```
Also, from the implementation, it looks like the array passed in should not be ALL mode states, but rather a list of nested mode states starting with the top-level mode and ending with a leaf mode?

Reply:
Thanks for that info. I am gonna look into that. I thought there was a single array with all mode_states.

Reply:
Yes you are right. I havent fully grapsed the modal design. But yes. There is not 1 entry per **mode_state,** but rather 1 entry per **mode**